backend/src/users/views.ts

```diff
@@ -75,89 +77,117 @@ userRouter.post(
   }
 );
 
-userRouter.delete("/:userid", useAuth, async (req: Request, res: Response) => {
-  // #swagger.tags = ['Users']
-  attempt(res, 200, () => userController.deleteUser(req.params.userid));
-  socketNotify("/users");
-});
-
-userRouter.put("/:userid", useAuth, async (req: Request, res: Response) => {
-  // #swagger.tags = ['Users']
-
-  // Do API call
-  attempt(res, 200, () =>
-    userController.updateUser(req.params.userid, req.body)
-  );
-  socketNotify(`/users/${req.params.userid}`);
-});
-
-userRouter.get("/count", useAuth, async (req: Request, res: Response) => {
-  // #swagger.tags = ['Users']
-  attempt(res, 200, userController.getCountUsers);
-});
-
-userRouter.get("/", useAuth, async (req: Request, res: Response) => {
-  // #swagger.tags = ['Users']
-  const filter = {
-    firstName: req.query.firstName as string,
-    lastName: req.query.lastName as string,
-    nickname: req.query.nickname as string,
-    email: req.query.email as string,
-    role: req.query.role as userRole,
-    hours: req.query.hours ? parseInt(req.query.hours as string) : undefined,
-    status: req.query.status as UserStatus,
-    eventId: req.query.eventId as string,
-    attendeeStatus: req.query.attendeeStatus as EnrollmentStatus,
-  };
-
-  const sortQuery = req.query.sort as string;
-  const querySplit = sortQuery ? sortQuery.split(":") : ["default", "asc"];
-  const key = querySplit[0];
-  const order = querySplit[1] as Prisma.SortOrder;
-  const sort = {
-    key: key,
-    order: order,
-  };
-
-  const pagination = {
-    after: req.query.after as string,
-    limit: req.query.limit as string,
-  };
-
-  attempt(res, 200, () => userController.getUsers(filter, sort, pagination));
-});
-
-userRouter.get("/pagination", useAuth, async (req: Request, res: Response) => {
-  // #swagger.tags = ['Users']
-  attempt(res, 200, () => userController.getUsersPaginated(req));
-});
-
-userRouter.get("/search", useAuth, async (req: Request, res: Response) => {
-  // #swagger.tags = ['Users']
-  const { email, firstName, lastName, role, status, hours, nickname } =
-    req.body;
-  attempt(res, 200, () =>
-    userController.getSearchedUser(
-      req,
-      email,
-      firstName,
-      lastName,
-      role,
-      status,
-      hours,
-      nickname
-    )
-  );
-});
-
-userRouter.get("/sorting", useAuth, async (req: Request, res: Response) => {
-  // #swagger.tags = ['Users']
-  attempt(res, 200, () => userController.getUsersSorted(req));
-});
+userRouter.delete(
+  "/:userid",
+  authIfAdmin as RequestHandler,
+  async (req: Request, res: Response) => {
+    // #swagger.tags = ['Users']
+    attempt(res, 200, () => userController.deleteUser(req.params.userid));
+    socketNotify("/users");
+  }
+);
+
+userRouter.put(
+  "/:userid",
+  authIfAdmin as RequestHandler,
```

Comment:
everyone is able to update their account. From testing, you will notice that this doesn't work because if you login as a volunteer and trying to change your name or something, it fails